### ISSUE TITLE
morgan types are broken

### ISSUE THREAD

**Issue by TrejGun** [DefinitelyTyped/DefinitelyTyped#50076]
Is something wrong with CI or publishing infrastructure?

- Check the [Infrastructure status updates](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/44317) thread for known issues.
- Leave a message on the [DefinitelyTyped Gitter](http://gitter.im/DefinitelyTyped/DefinitelyTyped).

If you know how to fix the issue, make a pull request instead.

- [x] I tried using the `@types/morgan` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I understand that `@types` package versions [do not follow SemVer](https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library).
- [ ] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @peterblazejewicz @staticfunction @pscanf

Few days ago I noticed that types are broken
Here is a simple code which throws error

```ts
const app = express();
app.use(morgan("tiny"));
```

```
error TS2769: No overload matches this call.
  The last overload gave the following error.
    Argument of type 'Handler<Request<never, never, never, never>, Response<never, number>>' is not assignable to parameter of type 'RequestHandlerParams<ParamsDictionary, any, any, ParsedQs>'.
      Type 'Handler<Request<never, never, never, never>, Response<never, number>>' is not assignable to type 'RequestHandler<ParamsDictionary, any, any, ParsedQs>'.
        Types of parameters 'req' and 'req' are incompatible.
          Type 'Request<ParamsDictionary, any, any, ParsedQs>' is not assignable to type 'Request<never, never, never, never>'.
            Type 'ParamsDictionary' is not assignable to type 'never'.
```

i can add generics and it will fix the issue but as I understand this should not be necessary 

```ts
app.use(morgan<Request, Response>("tiny"));
```

**Comment by peterblazejewicz** [comment]
If the `morgan` is used mostly in contex of the Express, maybe we could swap the implementation detail to work 
flawlessly against Express, for now that is the opposite, optimizing for plain node support.

**Comment by peterblazejewicz** [comment]
This is to be patched in this PR, #50072 
Sorry for the invonvience, we should probably amend `morgan` test to be included in the `express` CI gate, thx!
